src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1328 @@
+     * @throws DriverException
+     */
+    public function setDesiredCapabilities($desiredCapabilities): self
+    {
+        if ($this->isStarted()) {
+            throw new DriverException('Unable to set desiredCapabilities, the session has already started');
+        }
+
+        if (is_array($desiredCapabilities)) {
+            $desiredCapabilities = new DesiredCapabilities($desiredCapabilities);
+        }
+
+        $this->desiredCapabilities = $desiredCapabilities;
+
+        return $this;
+    }
+
+    /**
+     * Sets the timeouts to apply to the webdriver session
+     *
+     * @param array $timeouts The session timeout settings: Array of {script, implicit, page} => time in milliseconds
+     * @throws DriverException
+     * @api
+     */
+    public function setTimeouts(array $timeouts): void
+    {
+        $this->timeouts = $timeouts;
+
+        if ($this->isStarted()) {
+            $this->applyTimeouts();
+        }
+    }
+
+    /**
+     * Gets the final desired capabilities (as sent to Selenium).
+     *
+     * @see http://code.google.com/p/selenium/wiki/DesiredCapabilities
+     *
+     * @api
+     */
+    public function getDesiredCapabilities(): array
+    {
+        return $this->desiredCapabilities->toArray();
+    }
+
+    /**
+     * Globally press a key i.e. not typing into an element.
+     *
+     * @throws DriverException
+     * @api
+     */
+    public function globalKeyPress(string $char, ?string $modifier = null): void
+    {
+        $keyboard = $this->getWebDriver()->getKeyboard();
+        if ($modifier) {
+            $keyboard->pressKey($modifier);
+        }
+        $keyboard->pressKey($char);
+        if ($modifier) {
+            $keyboard->releaseKey($modifier);
+        }
+    }
+
+    /**
+     * Drag and drop an element by x,y pixels.
+     *
+     * @throws DriverException
+     * @api
+     */
+    public function dragBy(
+        #[Language('XPath')]
+        string $sourceXpath,
+        int $xOffset,
+        int $yOffset
+    ): void {
+        $source = $this->findElement($sourceXpath);
+        $this->getWebDriver()->action()->dragAndDropBy($source, $xOffset, $yOffset)->perform();
+    }
+
+    // </editor-fold>
+
+    // <editor-fold desc="Private Utilities">
+
+    /**
+     * @throws DriverException
+     */
+    private function getWebDriver(): RemoteWebDriver
+    {
+        if ($this->webDriver) {
+            return $this->webDriver;
+        }
+
+        throw new DriverException('Driver has not been started');
+    }
+
+    /**
+     * Detect and assign appropriate browser capabilities
+     *
+     * @see https://github.com/SeleniumHQ/selenium/wiki/DesiredCapabilities
+     */
+    private function initCapabilities(array $desiredCapabilities = []): DesiredCapabilities
+    {
+        // Build base capabilities
+        $browserName = $this->browserName;
+        if ($browserName && method_exists(DesiredCapabilities::class, $browserName)) {
```

Comment:
the fact that the method exists does not mean that it is a static factory method

Reply:
Seems that the only way is to keep a list of such methods. I realised the current code is wrong anyway, since it will look for a method called 'internet explorer' for example.